File: src/main/java/no/nav/aap/api/sts/STSSystemTokenTjeneste.java
package no.nav.aap.api.sts;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;

@Service
public class STSSystemTokenTjeneste implements SystemTokenTjeneste {

    private static final Logger LOG = LoggerFactory.getLogger(STSSystemTokenTjeneste.class);
    private final STSConnection connection;
    private SystemToken systemToken;

    public STSSystemTokenTjeneste(STSConnection connection) {
        this.connection = connection;
        LOG.info("System token {}", connection.refresh());
    }

    @Override
    public SystemToken getSystemToken() {
        if (systemToken == null || systemToken.isExpired(connection.getSlack())) {
            systemToken = connection.refresh();
        }
        return systemToken;
    }

    @Override
    public String ping() {
        return connection.ping();
    }

    @Override
    public String toString() {
        return getClass().getSimpleName() + "[connection=" + connection + ", systemToken=" + systemToken + "]";
    }

}

File: src/main/kotlin/no/nav/aap/api/oppslag/system/STSSystemTokenTjeneste.kt
package no.nav.aap.api.oppslag.system

import org.springframework.stereotype.Service

@Service
class STSSystemTokenTjeneste(private val adapter: STSWebClientAdapter) : SystemTokenTjeneste {
    private var systemToken: SystemToken? = null


    override fun getSystemToken(): SystemToken {
        if (systemToken == null || systemToken!!.isExpired(adapter.slack)) {
            systemToken = adapter.refresh()
        }
        return systemToken!!
    }
    override fun toString() = "${javaClass.simpleName} [adapter= $adapter, systemToken=$systemToken]"
}